### PR TITLE
Enable copy and select text in inbox messages

### PR DESCRIPTION
**Pull request by Ahmed-Naguib93**
refs :[MBL-17850](https://instructure.atlassian.net/browse/MBL-17850)
affects: Student, Teacher
release note: Enabled copy and select text in inbox messages

## Test plan:
- Open inbox messages.
- Select any message.
- Try to select or copy any text.
## Screenshots

https://github.com/user-attachments/assets/79eba56f-5f11-44f3-adc0-a7cb8a19d76a

## Checklist
- [ ] Tested on phone
- [ ] Tested in dark mode
- [ ] Tested in light mode

[MBL-17850]: https://instructure.atlassian.net/browse/MBL-17850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  </tbody>
</table>


#### Release Note: 
Enabled copy and select text in inbox messages

#### Affected Apps: Student, Teacher

[MBL-17850](https://instructure.atlassian.net/browse/MBL-17850)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.46% | 91.45% | 0.01%
Core/Core/Planner/CalendarEvent/Model/API/PostCalendarEventRequest.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarEvent/View/CustomFrequencyComponents/OccurrencesCountInputDialog.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarEvent/Model/Helpers/RecurrenceRule+SelectionDescription.swift | 43.33% | 43.33% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 988bd9e8e574b93fa451a64c20c1790c20275346
</p>

File: Core/Core/SwiftUIViews/SelectableTextViewWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        textView.textColor = UIColor(textColor)
+        textView.font = UIFont.scaledNamedFont(font)
+    }
+}
+
+extension SelectableTextViewWrapper {
+  public enum TextType {
+        case text(value: String)
+        case attributedText(value: AttributedString)
```
Comment (rh12) [inline Core/Core/SwiftUIViews/SelectableTextViewWrapper.swift:64]:
```suggestion
        case text(String)
        case attributedText(AttributedString)
```

File: Core/Core/SwiftUIViews/SelectableTextViewWrapper.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+public struct SelectableTextViewWrapper: UIViewRepresentable {
+    // MARK: - Properties
+    private let textType: TextType
+    private let font: UIFont.Name
+    private let textColor: Color
+
+    // MARK: - Init
+    public init(
+        textType: TextType,
```
Comment (rh12) [inline Core/Core/SwiftUIViews/SelectableTextViewWrapper.swift:29]: IMHO this init could be replaced with one expecting a `text: String` and another expecting `attributedText: AttributedString`. Those could setup the `textType` property, and `TextType` could be private.
But if you rather prefer it this way that's also ok.

**Review by rh12 — CHANGES_REQUESTED** [review]
It works as expected.

Some points of improvement:
1. Dragging the selection does not scroll the view, so user needs to scroll and adjust selection repeatedly. (Minor issue, may not worth the effort)
2. It would be good to have the "Select All" button to be shown besides the "Copy", etc. Because of the scrolling/reselecting limitation above, without this it's rather annoying to try to select all text (which was the way it worked before)

**Review by vargaat — APPROVED** [review]
QA+1

**Pull request merged**